File: 2021/12/14/main.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rangeSumBST(self, root: Optional[TreeNode], low: int, high: int) -> int:
        self.s = 0
        if root == None:
            return 0

        if root.val < low:
            self.s = self.rangeSumBST(root.right, low, high)
        elif root.val > high:
            self.s = self.rangeSumBST(root.left, low, high)
        else:
            self.s = root.val
            self.s += self.rangeSumBST(root.left, low, high)
            self.s += self.rangeSumBST(root.right, low, high)
        
        return self.s
        

File: 2021/12/14/main.go
package main

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func rangeSumBST(root *TreeNode, low int, high int) int {
	s := 0
	if root == nil {
		return 0
	}

	if root.Val < low {
		s = rangeSumBST(root.Right, low, high)
	} else if root.Val > high {
		s = rangeSumBST(root.Left, low, high)
	} else {
		s = root.Val
		s += rangeSumBST(root.Left, low, high)
		s += rangeSumBST(root.Right, low, high)
	}

	return s
}
